File: scripts/extract-sqlite.py
import sqlite3
import json
import sys
from datetime import datetime
import os

if len(sys.argv) < 2:
    print("❌ Please provide path to SQLite database")
    print("Usage: python extract-sqlite.py <path-to-db>")
    sys.exit(1)

db_path = sys.argv[1]

if not os.path.exists(db_path):
    print(f"❌ File not found: {db_path}")
    sys.exit(1)

print(f"🔍 Opening database: {db_path}\n")

try:
    conn = sqlite3.connect(db_path)
    conn.row_factory = sqlite3.Row
    cursor = conn.cursor()
    
    # Get all tables
    cursor.execute("SELECT name FROM sqlite_master WHERE type='table' ORDER BY name")
    tables = [row[0] for row in cursor.fetchall()]
    
    print(f"📊 Found {len(tables)} tables:\n")
    for table in tables:
        print(f"  - {table}")
    
    export_data = {
        "exportDate": datetime.now().isoformat(),
        "sourceFile": db_path,
        "tables": {}
    }
    
    print("\n📦 Extracting data...\n")
    
    for table in tables:
        try:
            cursor.execute(f"SELECT * FROM {table}")
            rows = cursor.fetchall()
            export_data["tables"][table] = [dict(row) for row in rows]
            print(f"  {table}: {len(rows)} rows")
        except Exception as e:
            print(f"  {table}: Error - {str(e)}")
    
    conn.close()
    
    # Save to JSON
    timestamp = datetime.now().isoformat().replace(':', '-').replace('.', '-')
    script_dir = os.path.dirname(os.path.abspath(__file__))
    backup_dir = os.path.join(os.path.dirname(script_dir), 'backups')
    os.makedirs(backup_dir, exist_ok=True)
    
    output_file = os.path.join(backup_dir, f"sqlite-export-{timestamp}.json")
    
    with open(output_file, 'w', encoding='utf-8') as f:
        json.dump(export_data, f, indent=2, ensure_ascii=False)
    
    file_size = os.path.getsize(output_file) / 1024
    
    print(f"\n✅ Data exported successfully!")
    print(f"📁 Location: {output_file}")
    print(f"📊 Size: {file_size:.2f} KB\n")
    
except Exception as error:
    print(f"❌ Error: {str(error)}")
    sys.exit(1)

File: scripts/extract-sqlite.js
const sqlite3 = require('better-sqlite3');
const fs = require('fs');
const path = require('path');

const dbPath = process.argv[2];
if (!dbPath || !fs.existsSync(dbPath)) {
  console.error('❌ Please provide a valid path to the SQLite database file');
  console.error('Usage: node extract-sqlite.js <path-to-db-file>');
  process.exit(1);
}

console.log(`🔍 Opening database: ${dbPath}\n`);

try {
  const db = sqlite3(dbPath, { readonly: true });
  
  // Get all tables
  const tables = db.prepare("SELECT name FROM sqlite_master WHERE type='table' ORDER BY name").all();
  
  console.log(`📊 Found ${tables.length} tables:\n`);
  tables.forEach(t => console.log(`  - ${t.name}`));
  
  const exportData = {
    exportDate: new Date().toISOString(),
    sourceFile: dbPath,
    tables: {}
  };
  
  console.log('\n📦 Extracting data...\n');
  
  tables.forEach(table => {
    try {
      const rows = db.prepare(`SELECT * FROM ${table.name}`).all();
      exportData.tables[table.name] = rows;
      console.log(`  ${table.name}: ${rows.length} rows`);
    } catch (err) {
      console.log(`  ${table.name}: Error - ${err.message}`);
    }
  });
  
  db.close();
  
  // Save to JSON
  const timestamp = new Date().toISOString().replace(/[:.]/g, '-');
  const outputFile = path.join(__dirname, '..', 'backups', `sqlite-export-${timestamp}.json`);
  
  fs.writeFileSync(outputFile, JSON.stringify(exportData, null, 2), 'utf8');
  
  console.log(`\n✅ Data exported successfully!`);
  console.log(`📁 Location: ${outputFile}`);
  console.log(`📊 Size: ${(fs.statSync(outputFile).size / 1024).toFixed(2)} KB\n`);
  
} catch (error) {
  console.error('❌ Error:', error.message);
  process.exit(1);
}
